src/Tribe/Integrations/WPML/Meta.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * Translate post ids in Event meta data.
+ *
+ * @since TBD
+ */
+class Tribe__Events__Integrations__WPML__Meta {
+
+	/**
+	 * @since TBD
+	 *
+	 * @var Tribe__Events__Integrations__WPML__Meta
+	 */
+	protected static $instance;
+
+	/**
+	 * @since TBD
+	 *
+	 * @return Tribe__Events__Integrations__WPML__Meta
+	 */
+	public static function instance() {
+		if ( empty( self::$instance ) ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Translates post id in the Event meta data.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $value
+	 * @param int    $object_id
+	 * @param string $meta_key
+	 *
+	 * @return string The translated id for _EventOrganizerID & _EventVenueID.
+	 */
+	public function translate_post_id( $value, $object_id, $meta_key ) {
+		global $wpdb;
+
+		if ( '_EventOrganizerID' !== $meta_key &&
+			 '_EventOrganizerID_Order' !== $meta_key &&
+			 '_EventVenueID' !== $meta_key ) {
+			return $value;
+		}
+
+		$value = $wpdb->get_col( $wpdb->prepare(
+			"SELECT meta_value FROM $wpdb->postmeta WHERE post_id = %d AND meta_key = %s",
+			$object_id,
+			$meta_key
+		) );
+
+		if ( empty( $value ) ) {
+			return false;
+		}
+
+		$type = strpos( 'Organizer', $meta_key )
+			? Tribe__Events__Organizer::POSTTYPE
+			: Tribe__Events__Venue::POSTTYPE;
+
+		foreach ( $value as & $post_id ) {
+			if ( is_serialized( $post_id ) ) {
+				$array = unserialize( $post_id );
```

Comment:
Making sure the value from `unserialize` is an `array` will prevent issues when is not maybe, using `is_arrray` or casting the value to an array `(array) unserialize` should be enough

Reply:
Added casting